iModelCore/ecobjects/PublicApi/ECObjects/ECSchema.h

```diff
@@ -2929,6 +2929,20 @@ struct SearchPathSchemaFileLocater : IECSchemaLocater, RefCountedBase, NonCopyab
     ECOBJECTS_EXPORT static SearchPathSchemaFileLocaterPtr CreateSearchPathSchemaFileLocater(bvector<WString> const& searchPaths, bool includeFilesWithNoVerExt=false);
 };
 
+//=======================================================================================
+//! Reads schemas from a given set of Xml strings
+//=======================================================================================
+struct EXPORT_VTABLE_ATTRIBUTE StringSchemaLocater : IECSchemaLocater, NonCopyableClass
+{
+private:
+    bmap<bpair<SchemaKey, SchemaMatchType>, ECSchemaPtr> m_knownSchemas;
+    bmap<Utf8String, Utf8String> m_schemaStrings;

```

Comment:
You should be able to get by with one variable that holds the schema strings by their key.  As noted schema locaters should not cache the deserialized schema, that is the schema contexts job.